packages/gestalt/src/ListText.js

```diff
@@ -0,0 +1,30 @@
+// @flow strict
+import { Children, type Element, type Node } from 'react';
+import Text from './Text.js';
+import styles from './List.css';
+import { useColorScheme } from './contexts/ColorSchemeProvider.js';
+
+type Props = {|
+  text: string | Element<typeof Text>,
+|};
+
+export default function ListText({ text }: Props): Node {
+  if (!text) {
+    throw new Error(`Gestalt List is missing \`label\` prop or a \`text\` prop within List.Item.`);
+  }
+
+  const { name: colorSchemeName } = useColorScheme();
+
+  // If `text` is a Text component, we need to override any text colors within to ensure they all match
+  if (typeof text !== 'string' && Children.only(text)?.type.displayName === 'Text') {
```

Comment:
What happens if `typeof text !== 'string'` but also `Children.only(text).type.displayName !== 'Text'`? It looks like we'll return the `text` Node wrapped in Text, which I don't think we want. Should we throw in that condition?

Reply:
Flow should catch it but I i'll throw.

Reply:
Oh good point, we do have the stricter type here